src/Generator/TypeBuilder.php

```diff
@@ -293,35 +397,87 @@ protected function buildScalarCallback($callback, string $fieldName)
     }
 
     /**
-     * @param mixed $resolve
+     * Builds a resolver closure that contains the compiled result of user-defined
+     * expression and optionally the validation logic.
+     *
+     * Render example (no expression language):
      *
-     * @return GeneratorInterface
+     *      function ($value, $args, $context, $info) use ($globalVariables) {
+     *          return "Hello, World!";
+     *      }
+     *
+     * Render example (with expression language):
+     *
+     *      function ($value, $args, $context, $info) use ($globalVariables) {
+     *          return $globalVariables->get('mutationResolver')->resolve(["my_resolver", [0 => $args]]);
+     *      }
+     *
+     * Render example (with validation):
+     *
+     *      function ($value, $args, $context, $info) use ($globalVariables) {
+     *          $validator = {@see buildValidatorInstance}
+     *          return $globalVariables->get('mutationResolver')->resolve(["create_post", [0 => $validator]]);
+     *      }
+     *
+     * Render example (with validation, but errors are injected into the user-defined resolver):
+     * {@link https://github.com/overblog/GraphQLBundle/blob/master/docs/validation/index.md#injecting-errors}
+     *
+     *      function ($value, $args, $context, $info) use ($globalVariables) {
+     *          $errors = new ResolveErrors();
+     *          $validator = {@see buildValidatorInstance}
+     *
+     *          $errors->setValidationErrors($validator->validate(null, false))
+     *
+     *          return $globalVariables->get('mutationResolver')->resolve(["create_post", [0 => $errors]]);
+     *      }
+     *
+     * @param mixed $resolve
      *
      * @throws GeneratorException
      * @throws UnrecognizedValueTypeException
+     *
+     * @return GeneratorInterface|string
      */
     protected function buildResolve($resolve, ?array $validationConfig = null)
     {
         if (is_callable($resolve) && is_array($resolve)) {
             return Collection::numeric($resolve);
         }
 
-        $closure = Closure::new()
-            ->addArguments('value', 'args', 'context', 'info')
-            ->bindVar(TypeGenerator::GLOBAL_VARS);
+        if (EL::isStringWithTrigger($resolve)) {
+            $closure = Closure::new()
+                ->addArguments('value', 'args', 'context', 'info')
+                ->bindVar(TypeGenerator::GLOBAL_VARS);
 
-        // TODO (murtukov): replace usage of converter with ExpressionLanguage static method
-        if ($this->expressionConverter->check($resolve)) {
-            $injectErrors = ExpressionLanguage::expressionContainsVar('errors', $resolve);
+            $injectErrors = EL::expressionContainsVar('errors', $resolve);
 
             if ($injectErrors) {
                 $closure->append('$errors = ', Instance::new(ResolveErrors::class));
```

Comment:
Having no **validation config**, but **errors** in the Expression might be a sign of an error. Still passing an **ResolveErrors** object might hide this fact. If you pass nothing, you would probably realize this via type or missing arg error. I'm not very tied to this, but just for you to consider if you did not already. :)

Reply:
@akomm Didn't quite understand.

> Still passing an ResolveErrors object might hide this fact

What fact can it hide?

Reply:
*Having no validation config, but errors in the Expression might be a sign of an error.  Still passing an ResolveErrors object might hide this fact*